### PR TITLE
Task 12 branch

### PR DESCRIPTION
**Pull request by Sisha3342**
Completed task number 12. **IMAGES DOESN'T WORK**

### CODE REVIEW

File: TwutterApp/src/main/java/twutter/PostActionServlet.java [hunk 1]
```diff
@@ -1,57 +1,107 @@
...
-    private static PostsList posts = new PostsList();
+    private static PostsList posts = new PostsList(15);
 
     @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
         String parameterName = "id";
-        String postId = req.getParameter(parameterName);
 
+        String postId = req.getParameter(parameterName);
```
Comment (smiakchilo) [inline TwutterApp/src/main/java/twutter/PostActionServlet.java:29]: The `parameterName` is evidently a constant. Use this as a true constant `private static final String PARAMETER_ID = "id"`;

File: TwutterApp/src/main/java/twutter/PostActionServlet.java [hunk 2]
```diff
@@ -1,57 +1,107 @@
...
         Post postToReturn = posts.getPost(postId);
 
         resp.getWriter().print((new Gson()).toJson(postToReturn));
     }
 
     @Override
-    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException {
```
Comment (smiakchilo) [inline TwutterApp/src/main/java/twutter/PostActionServlet.java:36]: ... and do not copy this likes-of-constant across methods)

File: TwutterApp/src/main/java/twutter/PostActionServlet.java [hunk 3]
```diff
@@ -1,57 +1,107 @@
...
         String[] reqUri = req.getRequestURI().split("/");
+        Map<String, String[]> map = req.getParameterMap();
 
         if (reqUri.length == 3 && reqUri[2].equals("search")) {
             Gson gson = new Gson();
 
-            resp.getWriter().print(posts.getPosts().stream().map(gson::toJson).
+            resp.getWriter().print(posts.getPosts(0, posts.getLength(), null).stream().map(gson::toJson).
+                    collect(Collectors.joining("\n")));
+        }
```
Comment (smiakchilo) [inline TwutterApp/src/main/java/twutter/PostActionServlet.java:53]: (code style) `} else` must be on a single line

File: TwutterApp/src/main/java/twutter/PostActionServlet.java [hunk 4]
```diff
@@ -1,57 +1,107 @@
...
+        else if (map.containsKey("skip")) {
+            Gson gson = new Gson();
+
+            Filter.Builder filterBuilder = new Filter.Builder();
+
+            if (map.containsKey("author") && !map.get("author")[0].equals("")) {
```
Comment (smiakchilo) [inline TwutterApp/src/main/java/twutter/PostActionServlet.java:59]: `!map.get("author")[0].isEmpty()`

File: TwutterApp/src/main/webapp/resources/js/controller.js [hunk 5]
```diff
@@ -2,37 +2,21 @@ class Controller {
...
-                    }
-                    break;
-                default:
-                    break;
+        let filtersForm = new FormData(document.forms.filtersForm);
+
+        testPosts.getPage(0, testPosts.getLength(), filtersForm).then(function (response) {
+            let filtered;
+
+
```
Comment (MaksimViarenich) [inline TwutterApp/src/main/webapp/resources/js/controller.js:10]: Minor. Remove one empty line

File: TwutterApp/src/main/webapp/resources/js/View.js [hunk 6]
```diff
@@ -181,13 +175,13 @@ class View {
...
-                <input name="postHashtagsInput" type="text" placeholder="tags">
+                <input name="hashTags" type="text" placeholder="tags">
                 
                 <h2>Description</h2>
-                <textarea name="postDescriptionInput" rows="10"></textarea>
+                <textarea name="description" rows="10"></textarea>
                 
                 <h2>Image</h2>
-                <input name="postImageInput" type="file" accept=".jpg, .jpeg, .png">
+                <input name="photoLink" type="file" accept=".jpg, .jpeg, .png">
```
Comment (MaksimViarenich) [inline TwutterApp/src/main/webapp/resources/js/View.js:184]: Thanks for clear naming)

**Pull request merged**